.github/workflows/build.yml

```diff
@@ -0,0 +1,163 @@
+# SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+#
+# SPDX-License-Identifier: MIT
+
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false #true
+      matrix:
+        os:
+          - ubuntu-22.04
+          - windows-2022
+          - macos-12
+        build_type:
+          - Debug
+          #- Release
+        client_only:
+          - client_only
+          - client_and_ui
+        qt:
+          - version: "5.15.2"
+            requested: "5.15"
+          - version: "6.5.3"
+            requested: "6.5.*"
+            modules: qtshadertools
```

Comment:
Come to think of it, I don't think that will quite work for Windows PowerShell, but this will:

```yaml
- run: >
    ${{ runner.os == 'windows'
    && '"QT_VERSION=$(qtpaths --qt-version)" >> $env:GITHUB_ENV' }}
    || 'echo "QT_VERSION=$(qtpaths --qt-version)" >> "$GITHUB_ENV"' }}

Reply:
AppImage can be created just fine by using qt plugin of Linux deploy, still it's too slow for regular builds.

Qt version is perfectly fine this way, as we might want to have more versions to test regressions or behavior changes. The version text is just text, what is used is the requested one, this is from the Qt git hub actions  page, which I might change in future to look a bit more like KDDockWidget, but there I create a much more complex setup so for now this is enough.